pkg/instancetype/instancetype.go

```diff
@@ -421,17 +443,57 @@ func (m *methods) getPreferenceSpecRevision(revisionName string, namespace strin
 	}
 }
 
+func (m *methods) getControllerRevisionByInformer(revisionName, namespace string) (*appsv1.ControllerRevision, error) {
```

Comment:
It might be helpful to use `types.NamespacedName` here?

https://pkg.go.dev/k8s.io/apimachinery/pkg/types#NamespacedName